test/test_luaunit.lua

```diff
@@ -2200,6 +2217,21 @@ TestLuaUnitExecution = { __class__ = 'TestLuaUnitExecution' }
         lu.assertEquals( runner.result.notPassed[1].status, lu.NodeStatus.FAIL  )
     end
 
+    function TestLuaUnitExecution:testWithCount()
+        local runner = lu.LuaUnit.new()
+        runner:setOutputType( "NIL" )
+
+        runner:runSuite( '--count', '5',
+                         'MyTestWithErrorsAndFailures.testOk')
+        lu.assertEquals( runner.exeCount, 5 )
+        lu.assertEquals( runner.currentCount, 5 )
+
+        runner:runSuite( '--count', '5',
+                         'MyTestWithErrorsAndFailures.testWithFailure1')
+        -- check if the current iteration got reflected in the failure message
+        lu.assertEquals( runner.result.failureCount, 1 )
+        lu.assertStrContains(runner.result.failures[1].msg, "iteration: 1")
+    end
```

Comment:
We need to check also how it behaves if the test starts failing at 3 iterations.